tests/e2e/scale/noobaa/test_scale_obc_creation_repsin_noobaa_pods.py

```diff
@@ -0,0 +1,118 @@
+from ocs_ci.framework.pytest_customization.marks import vsphere_platform_required
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def teardown(request):
+    def finalizer():
+        scale_noobaa_lib.cleanup(constants.OPENSHIFT_STORAGE_NAMESPACE)
+
+    request.addfinalizer(finalizer)
+
+
+@scale
+class TestScaleOCBCreation(E2ETest):
+    """
+    OBC scale creation, creating up to max support number of OBCs.
+    OBCs are created in the Multicloud Object Gateway and
+    Ceph Object Gateway (RGW)
+
+    """
+
+    namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+    scale_obc_count = 1000
+    num_obc_batch = 50
+
+    @pytest.mark.parametrize(
+        argnames=["pod_name", "sc_name"],
+        argvalues=[
+            pytest.param(
+                *["noobaa-core", constants.NOOBAA_SC],
+                marks=[
+                    pytest.mark.polarion_id("OCS-2645"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-db", constants.NOOBAA_SC],
+                marks=[
+                    pytest.mark.polarion_id("OCS-2646"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-core", constants.DEFAULT_STORAGECLASS_RGW],
+                marks=[
+                    vsphere_platform_required,
+                    pytest.mark.polarion_id("OCS-2647"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-db", constants.DEFAULT_STORAGECLASS_RGW],
+                marks=[
+                    vsphere_platform_required,
+                    pytest.mark.polarion_id("OCS-2648"),
+                ],
+            ),
+        ],
+    )
+    def test_scale_rgw_obc_creation(self, tmp_path, pod_name, sc_name, timeout=60):
+        """
+        OBC creation using RGW storage class
+        This test case only runs on vSphere cluster deployment
+        """
+
+        log.info(
+            f"Start creating  {self.scale_obc_count} "
+            f"OBC in a batch of {self.num_obc_batch}"
+        )
+        for i in range(int(self.scale_obc_count / self.num_obc_batch)):
+            obc_dict_list = (
+                scale_noobaa_lib.construct_obc_creation_yaml_bulk_for_kube_job(
+                    no_of_obc=self.num_obc_batch,
+                    sc_name=sc_name,
+                    namespace=self.namespace,
+                )
+            )
+            # Create job profile
+            job_file = ObjectConfFile(
+                name="job_profile",
+                obj_dict_list=obc_dict_list,
+                project=self.namespace,
+                tmp_path=tmp_path,
+            )
+            # Create kube_job
+            job_file.create(namespace=self.namespace)
+            time.sleep(timeout * 5)
+
+            # Check all the OBCs reached Bound state
+            obc_bound_list = (
+                scale_noobaa_lib.check_all_obc_reached_bound_state_in_kube_job(
```

Comment:
Looks like the test waits here until all the OBCs are bound. is the intention of the test to respin noobaa pods after OBC creation? 
I don't think there is a lot of value addition by respinning noobaa pods *after* OBCs are created. It would be useful to respin pods *during* OBC creation.

Reply:
Here is the flow of the test case: create 1000 obcs --> verifying obcs are in Bound state --> Re-spin noobaa pods --> Ensure noobaa pods are re-created and running, and created obcs are in the same Bound state.
The purpose of this test case is to verify that noobaa pods get re-spinned and come back in running state.
If re-spining noobaa pods happen during obc creation, the purpose of the test case is different which is focusing on obc creation but not noobaa pods re-spinning.

Reply:
OBC being "bound" is at *kubernetes* mgmt plane and there won't be any change in the state of OBC with noobaa pod restart. 
1) Do we run IOs on all these OBCs? (that may be a good validation)
2) delete these OBCs and validate there are no leaks

These validations would add a lot of value to the current test flow.

Reply:
Added 100 pods with IO running.